drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/rulesengine/SessionStatsCollector.java

```diff
@@ -119,4 +124,9 @@ public void registerAsyncResponse(byte[] bytes) {
     public void registerClockAdvance(long amount, TimeUnit unit) {
         clockAdvanceCount++;
     }
+
+    public void registerBaseLevelMemory() {
+        System.gc(); // NOSONAR
```

Comment:
Although `System.gc()` doesn't guarantee that JVM will do Full GC, I put it because of the nature of the requirement (= want to record the base level memory). As far as I tested, we get the memory size after a Full GC.